File: Array/Easy/Merge Strings Alternately/merge_strings_alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        ptr = 0
        result = ""
        while ptr < len(word1) or ptr < len(word2):
            if ptr<len(word1):
                result += word1[ptr]
            if ptr<len(word2):
                result += word2[ptr]
            ptr += 1
        
        return result     
    
solution_class = Solution()
word1 = input("Enter word 1: ")
word2 = input("Enter word 2: ")
result = solution_class.mergeAlternately(word1, word2)
print(result)

File: Array/Easy/Merge Strings Alternately/merge_strings_alternately.java
import java.util.Scanner;

public class merge_strings_alternately {
    public static void main(String[] args) {
        Solution solution_class = new Solution();
        Scanner scanner = new Scanner(System.in);
        String word1, word2;
        
        System.out.print("Enter word1: ");
        word1 = scanner.next();

        System.out.print("Enter word2: ");
        word2 = scanner.next();

        scanner.close();

        String result = solution_class.mergeAlternately(word1, word2);
        System.out.println(result);

    }
}

class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder result = new StringBuilder();
        int ptr = 0;
        int len1 = word1.length();
        int len2 = word2.length();

        while (ptr < len1 || ptr < len2) {
            if (ptr < len1) {
                result.append(word1.charAt(ptr));
            }
            if (ptr < len2) {
                result.append(word2.charAt(ptr));
            }

            ptr++;
        }

        return result.toString();
    }
}